lib/ts/utils.ts

```diff
@@ -453,22 +455,68 @@ export const useOnMountAPICall = <T>(
     }
 };
 
-export function saveInvalidClaimRedirectPathInContext(userContext: any, invalidClaimRedirectPath: string) {
-    if (userContext["_default"] === undefined) {
-        userContext["_default"] = {};
-    }
-    if (userContext["_default"].redirectPath === undefined) {
-        userContext["_default"] = {
-            ...userContext["_default"],
-            invalidClaimRedirectPath,
-        };
+export const getFailureRedirectionInfo = async ({
+    invalidClaims,
+    overrideGlobalClaimValidators,
+    userContext,
+}: {
+    invalidClaims: ClaimValidationError[];
+    overrideGlobalClaimValidators?:
+        | ((globalClaimValidators: SessionClaimValidator[], userContext: any) => SessionClaimValidator[])
+        | undefined;
+    userContext: any;
+}): Promise<{ accessForbidden: boolean; redirectPath: string | undefined }> => {
+    const globalValidatorsMap = getGlobalClaimValidators({
+        overrideGlobalClaimValidators,
+        userContext,
+    }).reduce((map, validator) => {
+        map[validator.id] = validator;
+        return map;
+    }, {} as Record<string, SessionClaimValidator>);
+
+    const redirects = [];
+
+    for (const claim of invalidClaims) {
+        const validator = globalValidatorsMap[claim.validatorId];
+        if (validator.onFailureRedirection) {
+            const redirectPath = await validator.onFailureRedirection({ reason: claim.reason, userContext });
+            redirects.push(redirectPath);
+            if (redirectPath !== undefined) {
+                break;
+            }
+        }
     }
-}
+    return {
+        accessForbidden: invalidClaims.length > 0 && redirects.length === 0,
+        redirectPath: redirects.filter(Boolean)[0],
```

Comment:
This will return undefined if the `onFailureRedirection` returns an empty string.

Reply:
yes which would mean that access is forbidden and we dont look whats in redirectPath

Reply:
I don't think that's true. If the first claim returns an empty string then we return: 
`{ accessForbidden: false, redirectPath: undefined}`
This is not the right behaviour.